### PR TITLE
Ability to perform integration test on Jupyter notebooks

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

Ability to perform integration test on Jupyter notebooks

Part of #1218

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5076** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5076" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5076?utm_source=stack-comment).

File: crates/ruff/src/jupyter/snapshots/ruff__jupyter__notebook__test__import_sorting.snap [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  |
+1 | / from pathlib import Path
+2 | | import random
+3 | | import math
+4 | | from typing import Any
+  | |_^ I001
+5 |   import collections
+6 |   # Newline should be added here
+  |
+  = help: Organize imports
```
Comment (dhruvmanila) [inline crates/ruff/src/jupyter/snapshots/ruff__jupyter__notebook__test__import_sorting.snap:14]: Oh right. I'll need to update the printer to include cell number and use Jupyter index.

File: crates/ruff/src/test.rs [hunk 2]
```diff
@@ -32,14 +45,41 @@ pub(crate) fn test_resource_path(path: impl AsRef<Path>) -> std::path::PathBuf {
...
+    let mut source_kind = SourceKind::Jupyter(read_jupyter_notebook(&path)?);
+    let messages = test_contents(&mut source_kind, &path, settings);
+    let expected_notebook =
+        read_jupyter_notebook(&test_resource_path("fixtures/jupyter").join(expected))?;
+    if let SourceKind::Jupyter(notebook) = source_kind {
+        assert_eq!(notebook.cell_offsets(), expected_notebook.cell_offsets());
+        assert_eq!(notebook.index(), expected_notebook.index());
+        assert_eq!(notebook.content(), expected_notebook.content());
+    };
+    Ok(messages)
```
Comment (dhruvmanila) [inline crates/ruff/src/test.rs:71]: Should we use snapshots as we'll be asserting the content here itself?

Comment (konstin) [reply crates/ruff/src/test.rs:71]: You could move the ` assert_messages!(diagnostics);` here but it's fine either way

**Review by konstin — APPROVED** [review]
Looks good!

File: crates/ruff/src/test.rs [hunk 3]
```diff
@@ -219,7 +267,35 @@ fn print_diagnostics(diagnostics: Vec<Diagnostic>, file_path: &Path, source: &st
...
 
-    print_messages(&messages)
+    if source_kind.is_jupyter() {
+        print_jupyter_messages(&messages, &filename, source_kind)
+    } else {
+        print_messages(&messages)
+    }
+}
+
+pub(crate) fn print_jupyter_messages(
```
Comment (dhruvmanila) [inline crates/ruff/src/test.rs:277]: This seems a bit hacky to print cell information in the snapshot but just thought to put it out there to gain some feedback.

We could just avoid creating snapshots and test out using `assert_eq!` which is already being done for `cell_offsets`, `index` and `content`.

For reference, this is the commit which is isolated to this change (https://github.com/astral-sh/ruff/pull/5076/commits/f63103b20cd2fd6a0ad9ebbc2fa75fc0eb79f6f2)

\cc @charliermarsh @konstin

Comment (konstin) [reply crates/ruff/src/test.rs:277]: i like it!

Comment (charliermarsh) [reply crates/ruff/src/test.rs:277]: I trust @konstin on this one, not gonna review closely ;)

**Comment by dhruvmanila** [comment]
I'll add some more tests now that we've the ability to do so!

**Pull request merged**